### ISSUE TITLE
Subtitles

### ISSUE THREAD

**Issue by Oink70** [OpenShot/openshot-qt#4744]
**Describe the new feature:**
Support for external subtitles files.

**Describe the solution you'd like:**
- Importing standard formatted subtitle text files (Srt, vtt, which are mostly used and optionally sbv, sub, ttml, rt, scc), having them rendered in a customizable font, size, color and screen position.
- Generate subtitles from text entries while editing. (Voice to text is a bridge too far, I think) and
- Having the choice to render them into an export, or saved as separate subtitle file, based on the changes made while editing (timing).

**Describe alternatives you have considered:**
Manually creating png's for every line of text and manually positioning on the time line.

**Comment by chenlung** [comment]
I propose that it be re-opened. It's essential for foreign-language features or those with hearing impairments.

**Comment by JacksonRG** [comment]
I agree this is important. I was under the impression that our captioning system used a common format. But it looks like there's a slight variation.

Openshot:
```
00:00:09:266 --> 00:00:14:266
Sample Caption
```
.srt (via [wikipedia](https://en.wikipedia.org/wiki/SubRip))
```
4
00:03:18,608 --> 00:03:20,371
I guess I was wrong
```
and .vtt (via [wikipedia](https://en.wikipedia.org/wiki/WebVTT))
```
00:11.000 --> 00:13.000
<v Roger Bingham>We are in New York City
```

Would you consider the first bullet point solved if a `.` or `,` were both accepted as a separator for the last number in the timestamps?

@Oink70 Regarding the other two bullet points: (@chenlung If you have any experience in this area, I'd love to hear your input as well)
2) Currently in our daily build (not yet in the release version) the `+` button will create a new timestamp wherever the playhead (preview position) is on the timeline. While the cursor is on that line, pressing the `+` again will fill in the ending timestamp. (In the new playhead position, or 5 seconds later, if the playhead hasn't moved.)
3) When you say an option to save the caption text to a file, you mean a single caption file for the whole project?

**Comment by JacksonRG** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Openshot Exporting Standard Subtitle Files](https://trello.com/c/0g25P1BR/177-openshot-exporting-standard-subtitle-files)

**Comment by chenlung** [comment]
Hi @JacksonRG,

At the moment, my subtitle editor of choice is [Subtitle Horse](https://subtitle-horse.com), and I'm eagerly awaiting its upcoming version.

In relation to audio-to-text, [Subtitle Edit](https://github.com/SubtitleEdit/subtitleedit) has [Vosk](https://github.com/alphacep/vosk-api). However, it recently implemented experimental support for [Whisper](https://openai.com/blog/whisper). At the moment, I use [Happy Scribe](https://happyscribe.com) and tidy from there.

SRT ([sample](https://github.com/OpenShot/openshot-qt/files/9742886/SubRip.zip) and [specification](http://zuggy.wz.cz)) and VTT ([samples](https://github.com/OpenShot/openshot-qt/files/9742887/WebVTT.zip) and [specification](https://www.w3.org/TR/webvtt1)) indeed use commas and full stops respectively, as far as I know.

Nowadays, I tend to submit files as iTunes Timed Text (because it's frame-based). See the [specification](https://www.apple.com/euro/itunes/lp-and-extras/docs/iTunes_Package_Film_Specification5.0.pdf) and [samples](https://github.com/OpenShot/openshot-qt/files/9742879/iTunes.Timed.Text.zip).

Support for other formats might be a good idea (others might be better placed to answer which), though I'd say some of the legacy ones should take lower priority (being able to at least open and convert would be ideal). For example, see [samples](https://github.com/OpenShot/openshot-qt/files/9742876/DVD.Studio.Pro.zip) and the [specification](https://web.archive.org/web/20091028173024/http://geocities.com/McPoodle43/DVDMaestro/stl_format.html) for DVD Studio Pro.

I'm happy to provide some more samples (perhaps privately) and try to do some testing.

**Comment by chenlung** [comment]
Still valid, I believe.

**Comment by huangweicong112** [comment]
I hope to know when subtitle will be supported, I think it is realy important.

**Comment by KOYK** [comment]
A video editor without subtitle support lol
